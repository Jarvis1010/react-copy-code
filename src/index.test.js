import React from 'react';
import toJSON from 'enzyme-to-json';
import { mount } from 'enzyme';
import CodeBlock from './index';

const code = `
const greeting = (greet = 'Hello') => (name = 'World') => {
    return greet + ' ' + name + '!';
  };
`;

const JSXBlock = ({ code }) => <pre><code>{code}</code></pre>;

const stringBlock = `<pre><code>${code}</code></pre>`;

describe('Copy to code', () => {
  it('Renders component with JSX', () => {
    const wrapper = mount(<CodeBlock><JSXBlock code={code} /></CodeBlock>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Renders component with innerHtml', () => {
    const wrapper = mount(<CodeBlock innerHtml>{stringBlock}</CodeBlock>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Renders component with highlight', () => {
    const wrapper = mount(
      <CodeBlock highlight><JSXBlock code={code} /></CodeBlock>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('has a pre code block', () => {
    const wrapper = mount(<CodeBlock><JSXBlock code={code} /></CodeBlock>);
    expect(wrapper.find('pre code').exists()).toBe(true);
  });

  it('has code in a pre code block', () => {
    const wrapper = mount(<CodeBlock><JSXBlock code={code} /></CodeBlock>);
    expect(wrapper.find('pre code').text()).toBe(code);
  });

  //   it('has div wrapper', () => {
  //     const wrapper = mount(<CodeBlock><JSXBlock code={code} /></CodeBlock>);
  //     expect(wrapper.find('.clipWrapper').exists()).toBe(true);
  //   });
});
