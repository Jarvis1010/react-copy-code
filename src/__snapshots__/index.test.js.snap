// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy to code Renders component with JSX 1`] = `
<CodeBlock
  element=""
  highlight={false}
  innerHtml={false}
>
  <styled.div
    innerRef={[Function]}
  >
    <div
      className="sc-bdVaJa knUAsi"
    >
      <JSXBlock
        code="
const greeting = (greet = 'Hello') => (name = 'World') => {
    return greet + ' ' + name + '!';
  };
"
      >
        <pre>
          <code>
            
const greeting = (greet = 'Hello') =&gt; (name = 'World') =&gt; {
    return greet + ' ' + name + '!';
  };

          </code>
        </pre>
      </JSXBlock>
    </div>
  </styled.div>
</CodeBlock>
`;

exports[`Copy to code Renders component with highlight 1`] = `
<CodeBlock
  element=""
  highlight={true}
  innerHtml={false}
>
  <styled.div
    innerRef={[Function]}
  >
    <div
      className="sc-htpNat icoewt"
    >
      <JSXBlock
        code="
const greeting = (greet = 'Hello') => (name = 'World') => {
    return greet + ' ' + name + '!';
  };
"
      >
        <pre>
          <code>
            
const greeting = (greet = 'Hello') =&gt; (name = 'World') =&gt; {
    return greet + ' ' + name + '!';
  };

          </code>
        </pre>
      </JSXBlock>
    </div>
  </styled.div>
</CodeBlock>
`;

exports[`Copy to code Renders component with innerHtml 1`] = `
<CodeBlock
  element=""
  highlight={false}
  innerHtml={true}
>
  <styled.div
    innerRef={[Function]}
  >
    <div
      className="sc-bwzfXH jIggJF"
    >
      &lt;pre&gt;&lt;code&gt;
const greeting = (greet = 'Hello') =&gt; (name = 'World') =&gt; {
    return greet + ' ' + name + '!';
  };
&lt;/code&gt;&lt;/pre&gt;
    </div>
  </styled.div>
</CodeBlock>
`;
